src/libbson/CMakeLists.txt

```diff
@@ -248,10 +248,17 @@ set_target_properties (bson_shared PROPERTIES VERSION 0.0.0 SOVERSION 0)
 set_target_properties (bson_shared PROPERTIES OUTPUT_NAME "bson-${BSON_API_VERSION}" PREFIX "lib")
 
 if (ENABLE_APPLE_FRAMEWORK)
+
+   set (MONGOC_FRAMEWORK_BUNDLE_VERSION ${MONGOC_MAJOR_VERSION}.${MONGOC_MINOR_VERSION}.${MONGOC_MICRO_VERSION})
+   if ("${MONGOC_PRELEASE_VERSION}" STREQUAL "")
```

Comment:
Should this check include a "NOT"?

Reply:
Also, according to the [docs for CFBundleVersion](https://developer.apple.com/library/archive/documentation/General/Reference/InfoPlistKeyReference/Articles/CoreFoundationKeys.html#//apple_ref/doc/uid/20001431-102364)

> While developing a new version of your app, you can include a suffix after the number that is being updated; for example 3.1.3a1. The character in the suffix represents the stage of development for the new version. For example, you can represent development, alpha, beta, and final candidate, by d, a, b, and fc. The final number in the suffix is the build version, which cannot be 0 and cannot exceed 255. When you release the new version of your app, remove the suffix.

Right now, MONGOC_PRERELEASE_VERSION is "pre" on the master branch. Would this need another number between 0-255 appended?

Reply:
Yeah, it might. There documentation doesn't say much about what can go here. I was assuming that `pre` would be fine, or `rc0` would be fine. But no way to know without trying I guess. I sort of don't want this to become an unreasonable constraint on your version scheme. I suppose another option would be to just completely ignore the actual value of `MONGOC_PRERELEASE_VERSION` here, and just always tack on something like `pre0`. But that is sort of a shame if we were ever in a situation where we had meaningful things like `rc1` but didn't add them.

So, I don't know. I'm unclear on how much validation is performed for the suffix. It is implied that at least some is, since it says that the terminal number must be less than 255. It also says it can't be zero, so `rc0` wouldn't actually work I guess. I think I missed that on the first time reading through that.

What do you think?

Reply:
Although not ideal, I vote for always appending "pre1" unless there's some standard scheme.

Also, `MONGOC_PRERELEASE_VERSION` is only "pre" if there's any prerelease info. Normally, our full prerelease info includes a date and git hash. E.g. "1.14.0-20181015+git7e2e9caeba". As @rcsanchez97 helpfully explained to me, the reason `MONGOC_PRERELEASE_VERSION` is only "pre" is because the full version is included in the driver ismaster handshake string, and the full prerelease string would be too long.

But perhaps the full prerelease string is appropriate for the framework though, and the docs don't seem to prohibit it. If so, I think change line 254 to:

```
set (MONGOC_FRAMEWORK_BUNDLE_VERSION "${MONGOC_DIST_VERSION}pre1")
```

`MONGOC_DIST_VERSION` is the full `<major>.<minor>.<patch>-<date>+git<hash>` string.